elements/howto-menu/howto-menu.js

```diff
@@ -0,0 +1,216 @@
+ * such as a set of actions or functions. A menu is usually opened,
+ * or made visible, by activating a menu button, choosing an item in a menu
+ * that opens a sub menu, or by invoking a command, such as Shift + F10 in
+ * Windows, that opens a context specific menu.
+ *
+ * The element that opens the menu is referenced with aria-labelledby.
+ * TODO(devnook): Should it be aria-labelledby or aria-describedby?
+ *
+ * See: https://www.w3.org/TR/wai-aria-practices-1.1/#menu
+ */
+(function() {
+  /**
+   * Define key codes to help with handling keyboard events.
+   */
+  const KEYCODE = {
+    DOWN: 40,
+    UP: 38,
+    ESCAPE: 27,
+  };
+
+  class HowtoMenu extends HTMLElement {
+
+    /**
+     * The constructor does work that needs to be executed _exactly_ once.
+     */
+    constructor() {
+      super();
+      this._handleBlur = this._handleBlur.bind(this);
+      this._opener = document.getElementById(
+         this.getAttribute('aria-labelledby'));
+    }
+
+    static get observedAttributes() {
+      return ['aria-hidden'];
+    }
+
+
+    /**
+     * A getter for the first child which is a menuitem.
+     */
+    get firstMenuItem() {
+      return this.querySelector('[role^="menuitem"]:first-of-type');
+    }
+
+    /**
+     * A getter for the last child which is a menuitem.
+     */
+    get lastMenuItem() {
+      return this.querySelector('[role^="menuitem"]:last-of-type');
+    }
+
+    /**
+     * Moves the browser focus from one element to another.
+     * @param {Element} fromEl Element to have focus removed.
+     * @param {Element} toEl Element to get focus.
+     */
+    _moveFocus(fromEl, toEl) {
+      fromEl.setAttribute('tabindex', -1);
+      toEl.setAttribute('tabindex', 0);
+      toEl.focus();
+    }
+
+    /**
+     * Checks if a node is a 'menuitem', 'menuitemcheckbox' or 'menuitemradio'.
+     */
+    _isMenuItem(node) {
+      let ariaRoles = ['menuitem', 'menuitemcheckbox', 'menuitemradio'];
+      return ariaRoles.indexOf(node.getAttribute('role') > -1);
+    }
+
+    /**
+     * Returns true if the menu is currently hidden, false otherwise.
+     */
+    _isHidden() {
+      return this.getAttribute('aria-hidden') === 'true';
+    }
+
+    /**
+     * Returns a menu item coming after the one passed as an argument, or null.
+     */
+    _nextMenuItem(node) {
+      let next = node.nextElementSibling;
+      while (next) {
+        if (this._isMenuItem(node)) {
+          return next;
+        }
+        next = next.nextElementSibling;
+      }
+      return null;
+    }
+
+    /**
+     * Returns a menu item coming before the one passed as an argument, or null.
+     */
+    _previousMenuItem(node) {
+      let prev = node.previousElementSibling;
+      while (prev) {
+        if (this._isMenuItem(node)) {
+          return prev;
+        }
+        prev = prev.previousElementSibling;
+      }
+      return null;
+    }
+
+    /**
+     * Opens the menu.
+     */
+    _open() {
```

Comment:
I'm not sure I got it correctly - you want the body of the _open and _close to be moved back  to  attributeChangedCallback? They are already called from there, just with the 'hidden' property and not 'opened'.